src-cpp/app/modules/main/module.h

```diff
@@ -12,14 +11,15 @@
 
 namespace Modules::Main
 {
-class Module : public QObject, virtual public IModuleAccess
+class Module : public QObject, public IModuleAccess
 {
     Q_OBJECT
-    Q_INTERFACES(Modules::Main::IModuleAccess)
 
 private:
     bool m_moduleLoaded;
 
+    // FIXME: don't use raw pointers
+    // (should be either plain member, reference or smart pointer, depending on ownerhip)
```

Comment:
Raw pointers are bad because one cannot tell what is the ownership by just looking at the interface.

`View m_View` - I am the owner of View, it will be destroyed when I'm
`std::unique_ptr<View> m_View` - like above + it can be dynamically initialized or changed at runtime
`std::shared_ptr<View> m_View` - I am one of the owners, it will not be destroyed until I'm alive
`View& m_View` - I am not the owner of View, I just have reference to it, somebody who gave me that reference must ensure it will outlive me

`View* m_View` - it's hard to tell who is the owner, either I can be the owner by managing it (calling `new` in the constructor and `delete` in the destructor) or I can be provided with the pointer through the constructor (I need to remember to not `delete` it as I am not the owner in that case)

Reply:
`std::weak_ptr<View> m_view` - I'm not he owner.

Reply:
but again, `View* m_view = new View(this);` - is correct too. if some thread call slot from view and you delete it without `deleteLater()` its crashed.